src/routes/search-results.js

```diff
@@ -5,30 +5,83 @@ import {
   Topbar
 } from '../components';
 
-// TODO: Delete this. This is only a temporary solution to the handle click
-//   callback.
 function handleClick() {
-  console.log("I've been clicked! Bless this day.");
+  console.log("Search Results card clicked.");
 }
 
+/**
+ * Improvements:
+ *  - Put this function in a shared file b/c it's copied from Favorites screen.
+ */
+function searchResultsListToHtml(data) {
+  return data.map((space, i) => {
+    return <Link to={{ state: { data : space }, pathname: "/view-space" }} style={{ textDecoration: "none", color: "#000000" }}>
+      <StudySpaceCard
+        index={i}
+        data={space}
+        handleClick={handleClick}
+        hasRemove={false}
+      />
+    </Link>;
+  });
+}
+
+/**
+ * Improvements:
+ *  - Remove all hard-coded junk code.
+ */
 const SearchResults = (props) => {
     const [noResults, setNoResults] = useState(false);
+    // NOTE: Requires that component attribute is set in App Container to access
+    //   this variable
+    const { data, preferences } = props.location.state;
+
+    console.log(preferences);
+
+    var ambience = 0;
+    if (preferences.ambience === 1) {
+      ambience = "quiet";
+    } else if (preferences.ambience === 2) {
+      ambience = "med";
+    } else {
+      ambience = "loud";
+    }
+
+    var popularity = 0;
+    if (preferences.popularity === 1) {
+      popularity = "not";
+    } else if (preferences.popularity === 2) {
+      popularity = "med";
+    } else {
+      popularity = "high";
+    }
```

Comment:
you will need to change this mapping. Take a look at week 7's ambience and popularity component files. For ambience, quiet, med, loud should be 1, 2, 3, respectively (they should be ints). Similarly for popularity, not, med, high should be, 1, 2, 3, respectively. Otherwise, the icons for ambience and popularity will not show.

Reply:
oh ok. this mapping was configured to previous preferences object that used strings. will change rn.